dbt/include/clickhouse/macros/adapters.sql

```diff
@@ -109,6 +109,11 @@
 {% endmacro %}
 
 {% macro exchange_tables_atomic(old_relation, target_relation, obj_types='TABLES') %}
+
+  {%- if adapter.get_clickhouse_cluster_name() is not none and obj_types == 'TABLES' %}
+    {% do run_query("SYSTEM SYNC REPLICA "+target_relation.identifier) %}
```

Comment:
@genzgd @saurabhbikram 
Don't we need to run it on cluster?
https://clickhouse.com/docs/en/sql-reference/statements/system/#sync-replica

I guess it will run sync only on single shard

Reply:
Yeah I think you're right.  I'm not very familiar with that command but syncing on cluster seems safer.

Reply:
Yes we should definitely add it. Thanks for highlighting.

Reply:
I'll stick it in the new release